### PR TITLE
Add initial ImageLab project with drag and drop + dragging UI elements to the build pane initial implementation.

### PR DESCRIPTION
**Pull request by dilumdesilva**
## Description
This pull request contains 
1. The initial project of ImageLab
2. Base UI developed using JavaFX. 
3. README update with the relevant project build-steps.
4. Dragging UI shapes from UI toolbar to build pane initial implementation

## Type of change 
- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] This change requires a documentation update.

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

## Current Status 
Main Dashboard

### CODE REVIEW

**Comment by ivantha** [comment]
Please include a screenshot of the latest state in the PR description.

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+                <groupId>org.openjfx</groupId>
+                <artifactId>javafx-maven-plugin</artifactId>
+                <version>0.0.3</version>
+                <configuration>
+                    <mainClass>com.imagelab.App</mainClass>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+</project>
```
Comment (ivantha) [inline pom.xml:44]: Should keep a newline at the end of the file. Please fix this and all the other similar places.

File: src/main/java/com/imagelab/DashboardController.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
+package com.imagelab;
+
+import java.io.File;
+import java.io.IOException;
+import java.security.KeyStore;
+import java.util.List;
```
Comment (ivantha) [inline src/main/java/com/imagelab/DashboardController.java]: Unused imports.

File: src/main/java/com/imagelab/DashboardController.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+import javafx.scene.input.*;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.shape.Circle;
+
+public class DashboardController {
+
+    private final DataFormat circleFormat = new DataFormat("javafx.scene.shape.Circle");
+
+    @FXML
+    private AnchorPane buidlanchorpane;
```
Comment (ivantha) [inline src/main/java/com/imagelab/DashboardController.java]: Naming should be consistent and Camel case. Please fix it in other relevant places as well.

File: .gitignore [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+# *.iml
+# *.ipr
+
+
+# File-based project format
+*.iws
+
+# IntelliJ
+out/
+
```
Comment (ivantha) [inline .gitignore]: We should add target directory to your .gitignore and never commit it.

File: src/main/java/com/imagelab/App.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+        stage.setScene(scene);
+        stage.setResizable(false);
+        stage.show();
+    }
+
+    static void setRoot(String fxml) throws IOException {
+        scene.setRoot(loadFXML(fxml));
+    }
+
+    private static Parent loadFXML(String fxml) throws IOException {
```
Comment (ivantha) [inline src/main/java/com/imagelab/App.java:39]: Let's use doc comments for any method except an overridden one.

File: src/main/java/com/imagelab/App.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+import javafx.fxml.FXMLLoader;
+import javafx.scene.Parent;
+import javafx.scene.Scene;
+import javafx.stage.Stage;
+
+import java.io.IOException;
+
+/**
+ * JavaFX App
+ */
```
Comment (ivantha) [inline src/main/java/com/imagelab/App.java]: Doc comments should conclude with a period. Let's add this as one of our checkstyles rules.

File: src/main/java/com/imagelab/DashboardController.java [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+import java.util.List;
+
+import javafx.fxml.FXML;
+import javafx.scene.input.*;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.shape.Circle;
+
+public class DashboardController {
+
+    private final DataFormat circleFormat = new DataFormat("javafx.scene.shape.Circle");
```
Comment (ivantha) [inline src/main/java/com/imagelab/DashboardController.java]: Use constants, please.

File: src/main/java/com/imagelab/DashboardController.java [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+
+    @FXML
+    private AnchorPane buidlanchorpane;
+
+    @FXML
+    private Circle circleoperation;
+
+    @FXML
+    private void handleDragDetection (MouseEvent event){
+        Dragboard db = circleoperation.startDragAndDrop(TransferMode.COPY);
```
Comment (ivantha) [inline src/main/java/com/imagelab/DashboardController.java]: It is okay to call this "dragboard" as db could mean database as well. 
As a matter of fact, it is better and easier to keep the naming as much closer to the camel case version of the class name.

**Review by ivantha — CHANGES_REQUESTED** [review]
I got a few suggestions. 
- Let's configure Checkstyles for the project. 
- Try to use more comments (doc ones too!).

File: pom.xml (hunk 1 above)
Comment (dilumdesilva) [reply pom.xml:44]: Sure I will fix.

File: src/main/java/com/imagelab/DashboardController.java (hunk 2 above)
Comment (dilumdesilva) [reply src/main/java/com/imagelab/DashboardController.java]: I will remove them.

File: src/main/java/com/imagelab/DashboardController.java (hunk 3 above)
Comment (dilumdesilva) [reply src/main/java/com/imagelab/DashboardController.java]: Sure, will check everwhere and fix this type of errors

File: src/main/java/com/imagelab/App.java (hunk 5 above)
Comment (dilumdesilva) [reply src/main/java/com/imagelab/App.java:39]: Sure I will add them

**Review by ivantha — APPROVED** [review]

**Pull request merged**